PowerEditor/src/NppNotification.cpp

```diff
@@ -88,6 +88,10 @@ BOOL Notepad_plus::notify(SCNotification *notification)
 				prevWasEdit = false;
 			}
 
+			if (notification->modificationType & SC_MOD_CHANGEINDICATOR)
+			{
+				::InvalidateRect(notifyView->getHSelf(), NULL, FALSE);
+			}

```

Comment:
Hmm... I'm wondering if it will bring a performance issue.

Reply:
It seems ok for me.

Reply:
It does not bring a performance issue, because calls to `InvalidateRect` do not immediately trigger `WM_PAINT` messages. So calling `InvalidateRect` repeatedly in a loop does no harm.

I checked it out by inserting `OutputDebugString` calls into the code before the `InvalidateRect` calls and the `WM_PAINT` messages sent to Scintilla (in `ScintillaEditView.cpp`). There is one additional `WM_PAINT` message sent per `addHotSpot` call.

But `addHotSpot` calls are seldom, because they are only executed after view range changes or text modifications. I can imagine that `addHotSpot` will be optimized soon, so that it causes no indicator change notifications anymore as long as no URLs have been modified in the view. But this will not affect performance, it will only look better in the debug logs.

Another source of indicator change notifications is the `xmlMatchedTagsHighlighter`. This one does its work after each text cursor move, and also causes one more `WM_PAINT` message per text cursor move. It could be optimized too. And this would also have a close-to-none effect on performance.

What struck me while this tests was, that most `WM_PAINT` messages sent to Scintilla have as update area only _the current line_ or _the current line together with the line above_ or _the current line together with the line below_. This `WM_PAINT` messages occur, when the text cursor is moved and when the text cursor is blinking. And exactly this `WM_PAINT` messages hide most of the potential update problems we may have. I think, this is why the problem has not been seen before. So, whenever it comes to test update problems, I would set the blink rate of the text cursor to max first.

Reply:
@Uhf7 Thank you very much for the testing and very detail explanation!